pymedphys/_dicom/anonymise/api.py

```diff
@@ -334,32 +340,45 @@ def anonymise_directory(
 
     identifying_keywords: ``list``, optional
         If left as None, the default values for/list of identifying keywords are used
+
+    Returns
+    -------
+    ``list`` of anonymised file paths
     """
     dicom_dirpath = str(dicom_dirpath)
 
     dicom_filepaths = glob(dicom_dirpath + "/**/*.dcm", recursive=True)
     failing_filepaths = []
-    # errors = []
+    successful_filepaths = []
+    anon_filepaths = []
+    errors = []
 
     for dicom_filepath in dicom_filepaths:
         if output_dirpath is not None:
             relative_path = os.path.relpath(dicom_filepath, start=dicom_dirpath)
             output_filepath = os.path.join(output_dirpath, relative_path)
         else:
             output_filepath = None
-
-        anonymise_file(
-            dicom_filepath,
-            output_filepath=output_filepath,
-            delete_original_file=delete_original_files,
-            anonymise_filename=anonymise_filenames,
-            replace_values=replace_values,
-            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
-            delete_private_tags=delete_private_tags,
-            delete_unknown_tags=delete_unknown_tags,
-            replacement_strategy=replacement_strategy,
-            identifying_keywords=identifying_keywords,
-        )
+        try:
+            dicom_anon_filepath = anonymise_file(
+                dicom_filepath,
+                output_filepath=output_filepath,
+                delete_original_file=delete_original_files,
+                anonymise_filename=anonymise_filenames,
+                replace_values=replace_values,
+                keywords_to_leave_unchanged=keywords_to_leave_unchanged,
+                delete_private_tags=delete_private_tags,
+                delete_unknown_tags=delete_unknown_tags,
+                replacement_strategy=replacement_strategy,
+                identifying_keywords=identifying_keywords,
+            )
+            successful_filepaths.append(dicom_filepath)
+            anon_filepaths.append(dicom_anon_filepath)
+        except (AttributeError, LookupError, TypeError, OSError, ValueError) as error:
+            errors.append(error)
+            failing_filepaths.append(dicom_filepath)
```

Comment:
I'm not sure which code being added back in you are referring to.
It appeared that someone had intended to do something similar previously (the # errors = [] and putting the conditional removal of existing files in a separate block), but there was no try/error block.
I copied entire functions at a time while splitting up anonymise/ moving things out of the __init__.py
Its a fair amount of history trawling to figure out if that try/error was there previously and if so, when it got removed, presumably by accident.  Or perhaps it was removed on purpose because of changed requirements (maybe it's better to fail up fast instead of continuing).
I think having the original author review and discuss what the thinking is regarding fail fast vs. complete as much as possible would be very valuable here.

Reply:
The "blame" UI in GitHub is awesome for going through the history:
https://github.com/pymedphys/pymedphys/blame/87042b829b8ed33082396f99acfd51a06a99bbd7/pymedphys/_dicom/anonymise/__init__.py#L380-L410

Here are the pertinent commits:

* https://github.com/pymedphys/pymedphys/commit/b05327a28ec1024a7ff0c67388848af7b455c282
* https://github.com/pymedphys/pymedphys/commit/4f7eb9747a5f358a2b3872a991df006b31777e68